listenbrainz/webserver/views/playlist_api.py

```diff
@@ -0,0 +1,516 @@
+
+    return True if value == "true" else False
+
+
+def validate_playlist(jspf, require_title=True):
+    """
+        Given a JSPF dict, ensure that title is present and that if tracks are present
+        they have valid URIs or MBIDs specified. If any errors are found 400 is raised.
+
+        If a created_for field is found and the user is not an approved playlist bot,
+        then a 403 forbidden will be raised.
+    """
+
+    if require_title:
+        try:
+            title = jspf["playlist"]["title"]
+            if not title:
+                raise KeyError
+        except KeyError:
+            log_raise_400("JSPF playlist must contain a title element with the title of the playlist.")
+
+    if 'track' not in jspf:
+        return
+
+    for i, track in enumerate(jspf['track']):
+        try:
+            recording_uri = track["identifier"]
+            if not recording_uri:
+                raise KeyError
+        except KeyError:
+            log_raise_400("JSPF playlist track %d must contain an identifier element with recording MBID." % i)
+
+        if recording_uri.startswith(PLAYLIST_TRACK_URI_PREFIX):
+            recording_mbid = recording_uri[len(PLAYLIST_TRACK_URI_PREFIX):]
+        else:
+            log_raise_400("JSPF playlist track %d identifier must have the namespace '%s' prepended to it." %
+                          (i, PLAYLIST_TRACK_URI_PREFIX))
+
+        if not is_valid_uuid(recording_mbid):
+            log_raise_400("JSPF playlist track %d does not contain a valid track identifier field." % i)
+
+
+def serialize_jspf(playlist: Playlist):
+    """
+        Given a playlist, return a properly formated dict that can be passed to jsonify.
+        TODO: Add tests for newly added fields.
+              Add collaborators
+    """
+
+    pl = {"creator": playlist.creator,
+          "title": playlist.name,
+          "identifier": PLAYLIST_URI_PREFIX + str(playlist.mbid),
+          "date": playlist.created.replace(tzinfo=datetime.timezone.utc).isoformat(),
+    }
+    if playlist.description:
+        pl["annotation"] = playlist.description
+    if playlist.created_for_id:
+        pl['created_for'] = playlist.created_for
+
+    extension = {"public": "true" if playlist.public else "false"}
+    extension["creator_id"] = playlist.creator_id
+    if playlist.copied_from_id:
+        extension['copied_from'] = PLAYLIST_URI_PREFIX + str(playlist.copied_from_id)
+
+    pl["extension"] = {PLAYLIST_EXTENSION_URI: extension}
+
+    tracks = []
+    for rec in playlist.recordings:
+        tr = {"identifier": PLAYLIST_TRACK_URI_PREFIX + str(rec.mbid)}
+        if rec.artist_credit:
+            tr["creator"] = rec.artist_credit
+
+        if rec.release_name:
+            tr["album"] = rec.release_name
+
+        if rec.title:
+            tr["title"] = rec.title
+
+        extension = {"added_by": rec.added_by,
+                      "added_at": rec.created.replace(tzinfo=datetime.timezone.utc).isoformat()}
```

Comment:
I think this is wrong, if the date that comes out of the database isn't in UTC, then this will take the time at that local timezone and then return it as if it was that hour UTC.
You probably want `rec.created.astimezone(datetime.timezone.utc)` instead

Reply:
Fixed.